### PR TITLE
Add tr.gov.uyap.Editor

### PR DESCRIPTION
**Pull request by otuva**
- [X] I have read the [App Requirements][reqs] and [App Maintenance][maint] pages.
- [X] My pull request follows the instructions at [App Submission][submission].
- [ ] I am using only the minimal set of permissions. *(If not, please explain each non-standard permission.)*
I'm using socket=pcsc and socket=cups on top of standard permissions because the app supports printing documents and usage of smartcards
- [X] All assets referenced in the manifest are redistributable by any party.  If not, the unredistributable parts are using an extra-data source type.
- [X] I am an upstream contributor to the project. If not, I contacted upstream developers about submitting their software to Flathub. 
- [ ] I own the domain used in the application ID or the domain has a policy for delegating subdomains (e.g. GitHub, SourceForge).
- [X] Any additional patches or files have been submitted to the upstream projects concerned. *(If not, explain why.)*

[reqs]: https://github.com/flathub/flathub/wiki/App-Requirements
[maint]: https://github.com/flathub/flathub/wiki/App-Maintenance
[submission]: https://github.com/flathub/flathub/wiki/App-Submission

### CODE REVIEW

File: tr.gov.uyap.Editor.appdata.xml [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+      <image>https://i.imgur.com/8HP6DSf.png</image>
+    </screenshot>
+  </screenshots>
+
+  <url type="homepage">https://uyap.gov.tr/UYAP-Editor</url>
+
+  <releases>
+    <release version="5.4.6" date="2022-11-01" />
+  </releases>
+</component>
```
Comment (hfiguiere) [inline tr.gov.uyap.Editor.appdata.xml:49]: missing required `content_rating`

File: tr.gov.uyap.Editor.Template.desktop [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+Type=Application
+
+Name=UYAP Template Editor
+Name[tr]=UYAP Şablon Editör
+Comment=UYAP Template Editor is an application to display and create USF documents in UYAP system.
+Comment[tr]=Uyap Şablon Editör uygulaması UYAP sisteminde oluşturulmuş USF dokümanlarının görüntülenmesi için hazırlanmıştır.
+Categories=Office;Java;
+MimeType=application/usf;
+
+Icon=tr.gov.uyap.Editor
```
Comment (hfiguiere) [inline tr.gov.uyap.Editor.Template.desktop:12]: shouldn't this be a different icon?

File: tr.gov.uyap.Editor.json [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+    "command": "uyapeditor",
+    "separate-locales": false,
+    "sdk-extensions": [
+        "org.freedesktop.Sdk.Extension.openjdk11"
+    ],
+    "finish-args": [
+        "--share=network",
+        "--share=ipc",
+        "--socket=x11",
+        "--socket=wayland",
```
Comment (hfiguiere) [inline tr.gov.uyap.Editor.json:15]: does it support wayland?

File: tr.gov.uyap.Editor.json [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+            "name": "uyapeditor",
+            "buildsystem": "simple",
+            "build-commands": [
+                "mkdir /app/bin",
+                "ln -s /app/jre/bin/java /app/bin/java",
+                "install -Dm644 tr.gov.uyap.Editor.png /app/share/icons/hicolor/128x128/apps/tr.gov.uyap.Editor.png",
+                "install -Dm644 tr.gov.uyap.Editor.desktop /app/share/applications/tr.gov.uyap.Editor.desktop",
+                "install -Dm644 tr.gov.uyap.Editor.Template.desktop /app/share/applications/tr.gov.uyap.Editor.Template.desktop",
+                "install -Dm644 tr.gov.uyap.Editor.appdata.xml /app/share/appdata/tr.gov.uyap.Editor.appdata.xml",
+                "rm    tr.gov.uyap.Editor.png    tr.gov.uyap.Editor.desktop    tr.gov.uyap.Editor.Template.desktop    tr.gov.uyap.Editor.appdata.xml",
```
Comment (hfiguiere) [inline tr.gov.uyap.Editor.json:40]:
```suggestion
```

File: tr.gov.uyap.Editor.json [hunk 5]
```diff
@@ -0,0 +1,75 @@
...
+                "ar -x uyapeditor_5.4.6_amd64.deb data.tar.xz",
+                "tar -xf data.tar.xz ./usr",
+                "cp -r usr/share/* /app/share/",
+                "rm data.tar.xz uyapeditor_5.4.6_amd64.deb",
```
Comment (hfiguiere) [inline tr.gov.uyap.Editor.json:44]:
```suggestion
```

**Comment by hfiguiere** [comment]
bot, build tr.gov.uyap.Editor

File: tr.gov.uyap.Editor.Template.desktop (hunk 2 above)
Comment (otuva) [reply tr.gov.uyap.Editor.Template.desktop:12]: Apparently not, it's not a different app per se. Just launches the app in a different mode. The app itself ships with two different desktop files without functionality to change inside the app. In order to not lose any functionality, I followed the same. I can add a saturated version if you'd like

Comment (hfiguiere) [reply tr.gov.uyap.Editor.Template.desktop:12]: if that's upstream UX then so be it.

**Comment by hfiguiere** [comment]
bot, build tr.gov.uyap.Editor

File: tr.gov.uyap.Editor.appdata.xml [hunk 6]
```diff
@@ -0,0 +1,53 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<component type="desktop-application">
+  <id>tr.gov.uyap.Editor</id>
+
+  <name>UYAP Editor</name>
+  <summary>UYAP Document Editor is an application to display and create UDF documents in UYAP
+    system.</summary>
+
+  <metadata_license>CC-BY-4.0</metadata_license>
+  <project_license>LicenseRef-proprietary</project_license>
```
Comment (barthalion) [inline tr.gov.uyap.Editor.appdata.xml:10]: Is it redistributable? If not, the manifest should be changed to use `extra-data`

File: README.md [hunk 7]
```diff
@@ -0,0 +1,11 @@
+# tr.gov.uyap.Editor
```
Comment (barthalion) [inline README.md:1]: This file is unlikely to be ever read by a regular user.

File: tr.gov.uyap.Editor.appdata.xml (hunk 6 above)
Comment (otuva) [reply tr.gov.uyap.Editor.appdata.xml:10]: Yes it is redistributable

File: README.md (hunk 7 above)
Comment (otuva) [reply README.md:1]: Yeah probably no one will ever see that. But pasted manifest description just in case lol

**Pull request closed without merging**